lib/web/apiserver.go

```diff
@@ -5166,11 +5167,37 @@ func (h *Handler) authExportPublic(w http.ResponseWriter, r *http.Request, p htt
 		return
 	}
 
-	reader := strings.NewReader(authorities)
+	h.serveTextResponse(authorities, w, r)
+}
+
+// Given an HTTP request with a textual response body, convert it to the requested response type.
+func (h *Handler) serveTextResponse(responseBody string, w http.ResponseWriter, r *http.Request) {
+	// This does not support parameters, and just ignores them.
+	// Trim any parameters and leave just the actual type
+	// Ex. text/plain;format=fixed;q=0.4 -> text/plain
+	requestedContentType, _, _ := strings.Cut(r.Header.Get("Accept"), ";")
+
+	var responseContentType string
+	var reader io.ReadSeeker
+	switch requestedContentType {
+	case "application/json":
+		responseContentType = "application/json"
+		responseContent, err := json.Marshal(responseBody)
+		if err != nil {
+			h.log.WithError(err).Error("Failed to convert response to a JSON string.")
+		}
```

Comment:
Should we add a return here?

Reply:
Should I add a return _after_ sending a response? If I return an error here, then the caller would either have to send the same response anyway, or break backwards compatibility.

Maybe error handling isn't even an issue here. Is there any way the marshal function could fail when provided a string? Should I just drop the error?